ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1945,6 +1945,18 @@ public RelOptMaterialization getMaterializedViewForRebuild(String dbName, String
     }
   }
 
+  public List<RelOptMaterialization> getMaterialization(
+          String queryString, List<String> tablesUsed, HiveTxnManager txnMgr) throws HiveException {
+
+    List<RelOptMaterialization> materializedViews =
+            HiveMaterializedViewsRegistry.get().getRewritingMaterializedViews(queryString);
+    if (materializedViews.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return filterAugmentMaterializedViews(materializedViews, tablesUsed, txnMgr);
```

Comment:
refactored this method: instead of calling `filterAugmentMaterializedViews` iterate through the mvs got from the registry and call `HiveMaterializedViewUtils.isOutdatedMaterializedView` for each of them. If it returns `null` or `true` filter out.

The majority of the code in `getValidMaterializedViews` can be ignored in this scenario because:
* all boolean parameters of `getValidMaterializedViews` is false
* We pull the MVs from the registry so all of them must exists in the registry

In case of multiple HS2 instances the registry may not contains all the possible MVs for rewrite but we initiate the lookup using the registry. 
I would extend this sql text based rewrite functionality to able to lookup from the metastore in a follow up patch.